### ISSUE TITLE
Move connection owner admin notice

### ISSUE THREAD

**Issue by ebinnion** [Automattic/jetpack#15568]
In the JITM package, there is some code for a notice that is displaying when attempting to disconnect the connection owner. Where that notice is hooked, the [following comment was written](https://github.com/Automattic/jetpack/blob/1f4935e91f76082ba242aa5cc75c4dc2f9cf5b7f/packages/jitm/src/class-jitm.php#L96):

> // Not really a JITM. Don't know where else to put this :) .

I'd like to suggest that we move that to one of the classes in the connection package since, in any plugin that uses the connection package, we'd likely want that notice to be displayed.

**Comment by kbrown9** [comment]
Yes, this definitely needs to be moved to the connection package!

We identified some issues that make moving the  `JITM::delete_user_update_connection_owner_notice()` method a bit complex:

- The method obtains the connected users that have the `jetpack_disconnect` [capability](https://github.com/Automattic/jetpack/blob/master/packages/jitm/src/class-jitm.php#L219). That capability is set up in the Jetpack [class](https://github.com/Automattic/jetpack/blob/master/class.jetpack.php#L1070).

- The [method includes a call to the Jetpack API](https://github.com/Automattic/jetpack/blob/master/packages/jitm/src/class-jitm.php#L267), which is set up by the Jetpack plugin. We will probably need to create an API package.

- The method includes strings that can be translated. We’ve worked on a POC for allowing consuming plugins to set their textdomain in the packages [here](https://github.com/Automattic/jetpack/pull/14650), but we haven’t had a chance to finalize the approach and finish that work yet.

We have a task card for this here: 1146297891914257-as-1146698724415329
